server/src/test/java/io/seata/server/lock/LockManagerTest.java

```diff
@@ -62,6 +66,43 @@ public void acquireLock_failed(BranchSession branchSession1, BranchSession branc
         Assertions.assertFalse(lockManager.acquireLock(branchSession2));
     }
 
+    /**
+     * deadlock test.
+     *
+     * @param branchSession1 the branch session 1
+     * @param branchSession2 the branch session 2
+     * @throws Exception the exception
+     */
+    @ParameterizedTest
+    @MethodSource("deadlockBranchSessionsProvider")
+    public void deadlockTest(BranchSession branchSession1, BranchSession branchSession2) throws Exception {
+        LockManager lockManager = new MemoryLockManagerForTest();
+        final AtomicBoolean first = new AtomicBoolean();
+        final AtomicBoolean second = new AtomicBoolean();
```

Comment:
What's the first and second variable used for?

Reply:
Before find the deadlock bug, i use them to make sure two concurrent branchSession register process with different row key list, at least one process could success.
Now i split that test case to another method.

Reply:
Ok, you lost that test case, now add a new function `concurrentUseAbilityTest` to make use of these two variables.